service/api_task.go

```diff
@@ -231,7 +231,7 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if t.Requester == evergreen.MergeTestRequester && details.Status != evergreen.TaskSucceeded {
+	if t.Requester == evergreen.MergeTestRequester && details.Status != evergreen.TaskSucceeded && !t.Aborted {
```

Comment:
this handles the fact that aborting a task also makes it "fail." We don't want this failure to be considered a real failure in the commit queue and we don't want the entry to get dequeued